common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -2276,22 +2276,25 @@ bool ap_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std:
 
         char MACAddress[MAC_ADDR_SIZE] = {0};
         int status_code                = 0;
+        char vap_name[beerocks::message::IFACE_NAME_LENGTH] = {0};
         size_t numOfValidArgs[4]       = {0};
         FieldsToParse fieldsToParse[]  = {
             {NULL /*opCode*/, &numOfValidArgs[0], DWPAL_STR_PARAM, NULL, 0},
-            {NULL, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL, 0},
+            {(void*) vap_name, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL,
+             beerocks::message::IFACE_NAME_LENGTH},
             {(void *)MACAddress, &numOfValidArgs[2], DWPAL_STR_PARAM, NULL, sizeof(MACAddress)},
             {(void *)&status_code, &numOfValidArgs[3], DWPAL_INT_PARAM, "status_code=", 0},
             /* Must be at the end */
             {NULL, NULL, DWPAL_NUM_OF_PARSING_TYPES, NULL, 0}};
 
-        if (dwpal_string_to_struct_parse(buffer, bufLen, fieldsToParse, sizeof(MACAddress)) ==
-            DWPAL_FAILURE) {
+        if (dwpal_string_to_struct_parse(buffer, bufLen, fieldsToParse,
+                                         sizeof(fieldsToParse)) == DWPAL_FAILURE) {
```

Comment:
Not sure why, but I have a Deja Vu when I wrote this comment.
Anyway, setting the last argument to `sizeof(fieldsToParse)` is wrong.
According to DWPAL code, the last argument should is defined by this description:
`size_t userBufLen - The user total buffer size to hold parsed info from inside fieldsToParse`
This means that it actually should be the size of `vap_name + MACAddress + status_code`.

I see wrong usage of it on all DWPAL functions.

Reply:
I wondered about that as well...

Reply:
Replaced with the sum of the size of each element.